shell/common/api/electron_api_native_image_mac.mm

```diff
@@ -53,32 +54,69 @@
   }
 
   CGSize cg_size = size.ToCGSize();
-  base::ScopedCFTypeRef<CFURLRef> cfurl = base::mac::FilePathToCFURL(path);
-  base::ScopedCFTypeRef<QLThumbnailRef> ql_thumbnail(
-      QLThumbnailCreate(kCFAllocatorDefault, cfurl, cg_size, NULL));
-  __block gin_helper::Promise<gfx::Image> p = std::move(promise);
-  // we do not want to blocking the main thread while waiting for quicklook to
-  // generate the thumbnail
-  QLThumbnailDispatchAsync(
-      ql_thumbnail,
-      dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_LOW, /*flags*/ 0), ^{
-        base::ScopedCFTypeRef<CGImageRef> cg_thumbnail(
-            QLThumbnailCopyImage(ql_thumbnail));
-        if (cg_thumbnail) {
-          NSImage* result =
-              [[[NSImage alloc] initWithCGImage:cg_thumbnail
-                                           size:cg_size] autorelease];
-          gfx::Image thumbnail(result);
-          dispatch_async(dispatch_get_main_queue(), ^{
-            p.Resolve(thumbnail);
-          });
-        } else {
-          dispatch_async(dispatch_get_main_queue(), ^{
-            p.RejectWithErrorMessage("unable to retrieve thumbnail preview "
-                                     "image for the given path");
-          });
-        }
-      });
+
+  if (@available(macOS 10.15, *)) {
+    NSURL* nsurl = base::mac::FilePathToNSURL(path);
+    QLThumbnailGenerationRequest* request = [[QLThumbnailGenerationRequest
+        alloc]
+          initWithFileAtURL:nsurl
+                       size:cg_size
+                      scale:1
```

Comment:
Should this be the screen's scale factor? I'm not sure if the old code generates HiDPI image.